### PR TITLE
Support tls for self-signed certificate

### PR DESCRIPTION
**Pull request by xiyichan**
Signed-off-by: xiyichan <2863768433@qq.com>

### CODE REVIEW

**Comment by DragonDriver** [comment]
Hello, @xiyichan , welcome to che community and thx for the contribution. Lgtm for the code changes generally. Besides, can you help to complete the documentation of the newly-added parameter. You can find the documentation in `connections.connect`. Btw, maybe the title of this patch should be more detailed, since tls was already supported. How about "Support to configure credential for tls"?

**Comment by xiyichan** [comment]
> Lgtm
If it is a self-signed certificate, we need to specify the certificate when we use it.

File: pymilvus/orm/connections.py [hunk 1]
```diff
@@ -140,6 +140,10 @@ def connect(self, alias=DefaultConfig.DEFAULT_USING, user="", password="", **kwa
             * *password* (``str``) --
                 Optional and required when user is provided. The password corresponding to
                 the user.
+            * *secure* (``bool``) --
+                Optional. Default is false. If set to true, tls will be enabled.
+            * *cert_path* (``str``) --
+                Optional. If use a self-signed certificate, need to write the server.crt path.
```
Comment (aakejiang) [inline pymilvus/orm/connections.py:146]: Will it need to specify ca.crt if CA exists?

Comment (xiyichan) [reply pymilvus/orm/connections.py:146]: I run the example.py is not need to specify ca.crt.

Comment (aakejiang) [reply pymilvus/orm/connections.py:146]: You should also test the case with CA.

Comment (xiyichan) [reply pymilvus/orm/connections.py:146]: i should build a ca server? I don't know how to test the case with CA.

File: pymilvus/client/grpc_handler.py [hunk 2]
```diff
@@ -123,14 +127,26 @@ def _setup_grpc_channel(self):
                              ('grpc.keepalive_time_ms', 55000)]
                 )
             else:
-                creds = grpc.ssl_channel_credentials(root_certificates=None, private_key=None, certificate_chain=None)
+                if self._client_pem_path != "" and self._client_key_path != "" and self._ca_pem_path != "" \
```
Comment (DragonDriver) [inline pymilvus/client/grpc_handler.py:130]: Should these four parameters appear at the same time?

File: pymilvus/client/grpc_handler.py [hunk 3]
```diff
@@ -123,14 +127,26 @@ def _setup_grpc_channel(self):
...
+                        root_certificates = f.read()
+                    creds = grpc.ssl_channel_credentials(root_certificates, private_key, certificate_chain)
+                else:
+                    creds = grpc.ssl_channel_credentials(root_certificates=None, private_key=None,
+                                                         certificate_chain=None)
                 self._channel = grpc.secure_channel(
                     self._uri,
                     creds,
-                    options=[(cygrpc.ChannelArgKey.max_send_message_length, -1),
+                    options=(('grpc.ssl_target_name_override', self._server_name, ),
```
Comment (DragonDriver) [inline pymilvus/client/grpc_handler.py:145]: What if `server_name` here is empty?

Comment (xiyichan) [reply pymilvus/client/grpc_handler.py:145]: server_name can't be empty

File: pymilvus/client/grpc_handler.py (hunk 2 above)
Comment (xiyichan) [reply pymilvus/client/grpc_handler.py:130]: Yes, I forget add server_name .

**Pull request merged**